eng/pipelines/libraries/stress/http.yml

```diff
@@ -50,31 +53,45 @@ jobs:
 
   - bash: |
       cd '$(httpStressProject)'
+      export CLIENT_SHARE="$(shareRoot)/client/3.0"
+      export SERVER_SHARE="$(shareRoot)/server/3.0"
       export HTTPSTRESS_CLIENT_ARGS="$HTTPSTRESS_CLIENT_ARGS -http 3.0 -xops 10"
       export HTTPSTRESS_SERVER_ARGS="$HTTPSTRESS_SERVER_ARGS -http 3.0"
       docker-compose up --abort-on-container-exit --no-color
     displayName: Run HttpStress - HTTP 3.0
     condition: and(eq(variables['buildRuntime.succeeded'], 'true'), eq(variables['buildStress.succeeded'], 'true'))
-
   - bash: |
       cd '$(httpStressProject)'
+      export CLIENT_SHARE="$(shareRoot)/client/2.0"
+      export SERVER_SHARE="$(shareRoot)/server/2.0"
       export HTTPSTRESS_CLIENT_ARGS="$HTTPSTRESS_CLIENT_ARGS -http 2.0"
       export HTTPSTRESS_SERVER_ARGS="$HTTPSTRESS_SERVER_ARGS -http 2.0"
       docker-compose up --abort-on-container-exit --no-color
     displayName: Run HttpStress - HTTP 2.0
     condition: and(eq(variables['buildRuntime.succeeded'], 'true'), eq(variables['buildStress.succeeded'], 'true'))
-
   - bash: |
       cd '$(httpStressProject)'
+      export CLIENT_SHARE="$(shareRoot)/client/1.1"
+      export SERVER_SHARE="$(shareRoot)/server/1.1"
       export HTTPSTRESS_CLIENT_ARGS="$HTTPSTRESS_CLIENT_ARGS -http 1.1"
       export HTTPSTRESS_SERVER_ARGS="$HTTPSTRESS_SERVER_ARGS -http 1.1"
       docker-compose up --abort-on-container-exit --no-color
     displayName: Run HttpStress - HTTP 1.1
     condition: and(eq(variables['buildRuntime.succeeded'], 'true'), eq(variables['buildStress.succeeded'], 'true'))
 
+  - bash: |
+      sudo chown -R $(id -u):$(id -g) $(shareRoot)
+    displayName: Own shared dumps and files
+    condition: failed()
+  - publish: $(shareRoot)
+    artifact: DumpsLinux
+    condition: failed()
```

Comment:
I'm also planning to publish an artifact for my stress reporting tool in #57653.

In the end I want to publish a directory containing an xml for each stress run. I think we should synchronize our work:
- Have one mounted volume with two subdirectories: one for dumps, one for report xml-s
- Publish separate artifacts for the two

Reply:
Going back to the `shareRoot`. What I wanted was one directory on the agent machine, where I'd create the dumps in different subdirectories (1.1 / 2.0 / 3.0, client / server etc.) and published the whole thing as one artifact "Dumps". There have to be at least two (Linux / Windows) because you cannot publish one artifact name twice. Unless, we create a separate job to publish them both at the same time.

If you want/need to publish your artifacts separately, you'd need to put the file in completely different directory (parallel to my dumps share). Or put it into the same share as I do the dumps and cherry pick the files afterwards when publishing the artifacts (it would still be concurently published with the dumps, which doesn't hurt anything but is redundant)

Reply:
I also prefer to publish two artifacts for Linux/Windows (what I meant is single artifact per job).

Can't we create a structure like this?

```

$(Build.ArtifactStagingDirectory) 
               |
               |------share <== this is the mount root we mount under /share or C:\share
                        |
                        |---- reports
                        |
                        |
                        |---- dumps
                                |
                                |-----1.0
                                |      |---- client
                                |      |---- server
                                |
                                |-----2.0
                                |      |---- client
                                |      |----- server
                                |
                                |-----3.0
                                       |---- client
                                       |----- server
```

Then your Linux/Windows steps publish `$(Build.ArtifactStagingDirectory)/dumps` (which contains stuff under `/share/dumps/**`), while my step does it with  `(Build.ArtifactStagingDirectory)/reports`.

Does this make sense or I'm missing something?

Reply:
Looks good, I'll try that.

Reply:
Hmmm, how will I know (inside the container) where to put the dumps? I.e. what to set to `ENV COMPlus_DbgMiniDumpName="/share/coredump.%p"`. 
I'd need something like `ENV COMPlus_DbgMiniDumpName="/share/dumps/1.0/client/coredump.%p"` but at the point in time I don't know that unless I manually parse `$HTTPSTRESS_ARGS`.

Reply:
Maybe we can define a separate `HTTP_VERSION` variable instead of always overwriting `HTTPSTRESS_CLIENT_ARGS` and `HTTPSTRESS_SERVER_ARGS` and then somehow push down the dump location from `docker-compose.yml`?

Reply:
We would need version and client/server both, that's something what I was trying to say in https://github.com/dotnet/runtime/pull/58860#discussion_r710098148. And I'm hesitant to do...
It's either 2 different mounts (for you and for me separately) or 1 mount and 2 extra ARGs for the dockerfiles... Or 1 mount and 1 extra ARG with duplicated info...

Reply:
If it's simpler to go with two separate shares then let's do it that way, though I still think the the 2 artifacts shall be created from 2 separate directories under `$(Build.ArtifactStagingDirectory)`.

Reply:
I changed the share root to `$(Build.ArtifactStagingDirectory)` as you suggested, but I'm keeping the rest as-is, e.g. going with 2 shares.